src/Experiment/src/AssemblyBuilder.cs

```diff
@@ -1,27 +1,67 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using System.Reflection;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
 
 namespace System.Reflection.Emit.Experimental
 {
+
     public class AssemblyBuilder: System.Reflection.Assembly
     {
-        internal AssemblyBuilder() { }
+        private AssemblyName? _assemblyName;
+        private MetadataBuilder _metadata = new MetadataBuilder();
+        private IDictionary<string,ModuleBuilder> _moduleStorage = new Dictionary<string, ModuleBuilder>();
+
+        private AssemblyBuilder(AssemblyName name) 
+        {
+            _assemblyName = name;
+        }
 
         public void Save(string assemblyFileName)
         {
-            //New method
+            if (assemblyFileName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyFileName));
+            }
+            if (_assemblyName==null||_assemblyName.Name==null)
```

Comment:
```suggestion
            if (_assemblyName == null || _assemblyName.Name == null)
```
NIT: adding spaces in conditions and statements help with readability